pkg/storage/calico/policy_storage_test.go

```diff
@@ -384,20 +387,30 @@ func TestNetworkPolicyGuaranteedUpdate(t *testing.T) {
 	}}
 
 	for i, tt := range tests {
+		klog.Infof("Start to run test on tt: %+v", tt)
 		out := &calico.NetworkPolicy{}
 		selector := fmt.Sprintf("my_label == \"foo-%d\"", i)
 		if tt.expectNoUpdate {
 			selector = ""
 		}
 		version := storeObj.ResourceVersion
-		err := store.GuaranteedUpdate(ctx, tt.key, out, tt.ignoreNotFound, tt.precondition,
+		versionInt, err := strconv.Atoi(version)
+		if err != nil {
+			t.Errorf("#%d: failed to convert original version %s to int", i, version)
+		}
+		err = store.GuaranteedUpdate(ctx, tt.key, out, tt.ignoreNotFound, tt.precondition,
 			storage.SimpleUpdate(func(obj runtime.Object) (runtime.Object, error) {
 				if tt.expectNotFoundErr && tt.ignoreNotFound {
 					if policy := obj.(*calico.NetworkPolicy); policy.Spec.Selector != "" {
 						t.Errorf("#%d: expecting zero value, but get=%#v", i, policy)
 					}
 				}
+
 				policy := *storeObj
+				// Set correct resource name, don't update "non-existing" to "foo"
```

Comment:
I don't think I understand this change - is it needed?

Reply:
We are running multiple update tests by walking through a slice of test structures. Those tests basically act on  two keys, `non-existing` and `foo`, where `foo` object is already created in the store. 

We are passing a single `userUpdate` function pointer to `GuaranteedUpdate` that override the original object with stored object `foo`. If the original object is `non-existing`, the name of the object will be updated to `foo`  after calling `userUpdate`. This is not correct as `userUpdate` function should never change the name of the object.